packages/lib-classifier/src/hooks/useFreehandLineReductions.js

```diff
@@ -24,47 +15,37 @@ export default function useFreehandLineReductions() {
     }
   } = useStores()
 
-  const { loaded, caesarReductions } = useCaesarReductions(workflow.caesarReducer)
-
-  const points = caesarReductions?.consensusLines(frame)
-  const activeStepAnnotations = stepHistory?.latest.annotations
-
-  // We expect there to only be one
-  const [task] = findTasksByType('drawing')
+  // NOTE: useFreehandLineReductions() and useTranscriptionReductions() are called regardless of reducer key
+  if (workflow.caesarReducer !== 'machine-learnt') return
 
-  // We want to observe the marks array for changes, so pass that as a separate prop.
-  const marks = task?.marks
-
-  // find the corresponding annotation
-  const annotation = activeStepAnnotations.find(
-    annotation => annotation.task === task?.taskKey
-  )
+  const { loaded, caesarReductions } = useCaesarReductions(workflow.caesarReducer)
 
-  if (marks.length === 0 && points) {
-    task.activeTool.createMark({
-      frame: 0,
-      id: cuid(),
-      toolIndex: 0,
-      points: points[0],
-      pathX: points.pathX,
-      pathY: points.pathY
+  const caesarMark = caesarReductions?.findCurrentTaskMark({
+    stepKey: step.stepKey,
+    tasks: step.tasks,
+    frame,
+  })
+
+  let task = step?.tasks[caesarMark?.taskIndex]
+  let tool = task?.tools[caesarMark?.toolIndex]
+  let marks = tool?.marks;
+  let marksjson = marks?.toJSON()
+
+  if (marks && caesarMark && marksjson[caesarMark.markId] === undefined) {
```

Comment:
`marks` is a [Map](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map)

Reply:
```suggestion
  if (marks && caesarMark && !marks.has(caesarMark.markId)) {
```